modules/payments/modules/payments/paymentHistoryModal.js

```diff
@@ -0,0 +1,130 @@
+import React, { PureComponent, useState, useEffect } from 'react';
+import { View, Text, Modal, StyleSheet, Pressable, Alert, RefreshControl } from 'react-native';
+import { FlatList, TouchableOpacity } from 'react-native-gesture-handler';
+import { fetchPaymentHistory } from "./api";
+import { Dimensions } from 'react-native'
+let deviceWidth = Dimensions.get('window').width
+
+
+const PaymentHistoryModal = (props) => {
+
+    const [paymentHistory, setPaymentHistory] = useState([]);
+    const [modalVisible, setModalVisible] = useState(false);
+    const [refresh, setRefresh] = useState(false);
+
+    useEffect(() => {
+        get_payments();
+    }, [])
+
+    const get_payments = async () => {
+        setRefresh(true);
+        const res = await fetchPaymentHistory()
+        fetchPaymentHistory().then(res => {
+            setRefresh(false);
+            setPaymentHistory(res);
+        })
+    }
```

Comment:
I actually wanted to do that but the login module was not storing data there I'll look again though. but then I recalled our conversation on using something like local or encrypted storage because otherwise the token would go away on app restart if its not persisted! thoughts ?

Reply:
That would be a feature of the login module, we can skip that here for now. I'm going to register that for future improvements to the login module.